src/extensions/YAKL_fft.h

```diff
@@ -145,35 +154,63 @@ namespace yakl {
+          if constexpr (std::is_same_v<T,float>) {
+            plan_forward = new desc_single_t(n);
+            plan_inverse = new desc_single_t(n);
 
-          CHECK( static_cast<desc_double_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, batch) );
-          CHECK( static_cast<desc_double_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE,         idist) );
-          CHECK( static_cast<desc_double_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE,         odist) );
+            CHECK( static_cast<desc_single_t*>(plan_forward)->set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, batch) );
+            CHECK( static_cast<desc_single_t*>(plan_forward)->set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE,         idist) );
+            CHECK( static_cast<desc_single_t*>(plan_forward)->set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE,         odist) );
 
-          CHECK( static_cast<desc_double_t*>(plan_forward)->commit( sycl_default_stream() ) );
-          CHECK( static_cast<desc_double_t*>(plan_inverse)->commit( sycl_default_stream() ) );
-        }
+            CHECK( static_cast<desc_single_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, batch) );
+            CHECK( static_cast<desc_single_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE,         odist) );
+            CHECK( static_cast<desc_single_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE,         idist) );
+
+            CHECK( static_cast<desc_single_t*>(plan_forward)->commit( sycl_default_stream() ) );
+            CHECK( static_cast<desc_single_t*>(plan_inverse)->commit( sycl_default_stream() ) );
+          } else if constexpr (std::is_same_v<T,double>) {
+            plan_forward = new desc_double_t(n);
+            plan_inverse = new desc_double_t(n);
+
+            CHECK( static_cast<desc_double_t*>(plan_forward)->set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, batch) );
+            CHECK( static_cast<desc_double_t*>(plan_forward)->set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE,         idist) );
+            CHECK( static_cast<desc_double_t*>(plan_forward)->set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE,         odist) );
+
+            CHECK( static_cast<desc_double_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, batch) );
+            CHECK( static_cast<desc_double_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE,         idist) );
+            CHECK( static_cast<desc_double_t*>(plan_inverse)->set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE,         odist) );
+
+            CHECK( static_cast<desc_double_t*>(plan_forward)->commit( sycl_default_stream() ) );
+            CHECK( static_cast<desc_double_t*>(plan_inverse)->commit( sycl_default_stream() ) );
+          }
+        #endif
```

Comment:
You can write this part much simpler. Write a to_precision struct as in the double batched FFT library:
```c++
template <typename T> struct to_precision;
template <> struct to_precision<float> {
    static constexpr auto value = oneapi::mkl::dft::precision::SINGLE;
};
template <> struct to_precision<double> {
    static constexpr auto value = oneapi::mkl::dft::precision::DOUBLE;
};
template <typename T> inline constexpr auto to_precision_v = to_precision<T>::value;
```
Then write
```
using fft_desc_t = oneapi::mkl::dft::descriptor<to_precision_v<T>, oneapi::mkl::dft::domain::REAL>;
```
and you don't need the if selection statement anymore. Moreover, if you remove the if-selection here then you don't need a heap allocation but you can allocate the plan on the stack.